src/snowflake/cli/api/metrics.py

```diff
@@ -213,15 +248,57 @@ def counters(self) -> Dict[str, int]:
         # return a copy of the original dict to avoid mutating the original
         return self._counters.copy()
 
+    @property
+    def num_spans_past_depth_limit(self) -> int:
+        return len(
+            [
+                span
+                for span in self._completed_spans
+                if span.span_depth > self.SPAN_DEPTH_LIMIT
+            ]
+        )
+
+    @property
+    def num_spans_past_total_limit(self) -> int:
+        return max(
+            0,
+            # we don't count the spans past the depth limit since they would not be reported
```

Comment:
why is this the case? i thought we wanted to report on the full number of spans opened by the app?

Reply:
+1, these limits are just guardrails.

Reply:
In my head these two are measuring which of the two guardrails we hit: total for horizontal and depth for vertical, and it would help investigation if they were isolated

We can infer the full number by summing the subtree counts (or introduce an explicit variable), and the semantics of having the spans exceeding depth limit also contribute to the total limit seems weird to me:
e.g. we have a path of 6 spans nested within each other, we then would say that 1 exceeded depth limit and 1 exceeded total limit, when we only report 5 of them and our total limit is 100 (which was not hit)

Happy to discuss further since I might be missing something though

Reply:
> we then would say that 1 exceeded depth limit and 1 exceeded total limit
woudn't it be 0 exceeded total limit, since 6 < 100?

Reply:
Discussed offline, changes made here: https://github.com/snowflakedb/snowflake-cli/pull/1835/commits/a2982b84a7b3d99baf3a70c8d334b84516859270